File: bongf/java/src/ch12_implementation_qeustions/Q7_LuckyStraight.java
package ch12_implementation_qeustions;

import java.util.Scanner;

public class Q7_LuckyStraight {
    public static void main(String[] args) {
        Q7_LuckyStraight ls = new Q7_LuckyStraight();
//        System.out.println(ls.solution());
        System.out.println(ls.dongbin());
    }

    public String solution() {
        Scanner sc = new Scanner(System.in);
        String[] ss = sc.nextLine().split("");

        int sum = 0;
        for (int i = 0; i < (ss.length/2); i++) {
            sum += Integer.parseInt(ss[i]);
        }

        for (int i =(ss.length/2); i < ss.length; i++) {
            sum -= Integer.parseInt(ss[i]);
        }

        if(sum==0) {
            return "LUCKY";
        }
        return "READY";
    }

    // charAt 활용
    public String dongbin() {
        Scanner sc = new Scanner(System.in);
        String str = sc.next();

        int sum = 0;
        for (int i = 0; i < (str.length()/2); i++) {
            sum += str.charAt(i) - '0';
        }

        for (int i =(str.length()/2); i <str.length(); i++) {
            sum -= str.charAt(i) - '0';
        }

        if(sum==0) {
            return "LUCKY";
        }
        return "READY";
    }
}

/*
123402
LUCKY
7755
READY
 */

File: bongf/python/dongbinbook/ch12_implementation_questions/q7_lucky_straight.py
# 절반 나눈다는 의미를 빼먹음 (앞에서부터 개수에 상관없이 값만 절반만 되면 되는 것으로 계산)
def solution() :
    data = list(map(int, input()))

    t = 0 
    for d in data :
        t += d 
    
    t /= 2 
    s = 0
    for d in data :
        s += d 
        if ( s == t) :
            return "LUCKY"
        if ( s > t) :
            return "READY"


print(solution())

def dongbin() :
    n = input()
    length = len(n)
    sum = 0 

    for i in range(length//2) :
        sum += int(n[i])
    
    for i in range(length//2, length):
        sum -= int(n[i])
    
    if sum == 0 :
        return "LUCKY"
    return "READY"

'''
123402
LUCKY
7755
READY 
'''
